client/src/main/scala/com/dwolla/cloudflare/ZoneSettingsClient.scala

```diff
@@ -0,0 +1,64 @@
+package com.dwolla.cloudflare
+
+import _root_.io.circe._
+import _root_.io.circe.generic.auto._
+import _root_.io.circe.syntax._
+import cats._
+import cats.data.Validated._
+import cats.data._
+import cats.effect._
+import cats.implicits._
+import fs2._
+import _root_.org.http4s.circe._
+import com.dwolla.cloudflare.ZoneSettingsClient._
+import com.dwolla.cloudflare.domain.dto.ZoneSettingsDTO
+import com.dwolla.cloudflare.domain.model.ZoneSettings._
+import com.dwolla.cloudflare.domain.model._
+import org.http4s.Method._
+import org.http4s.Uri
+import org.http4s.client.dsl.Http4sClientDsl
+
+import scala.concurrent.ExecutionContext
+
+trait ZoneSettingsClient[F[_]] {
+  def updateSettings(domain: Zone): Stream[F, ValidatedNel[Throwable, Unit]]
+}
+
+object ZoneSettingsClient {
+  type CloudflareSettingFunction = Zone ⇒ ZoneId ⇒ (Uri, Json)
+
+  val setTlsLevel: CloudflareSettingFunction = zone ⇒ zoneId ⇒ (BaseUrl / "zones" / zoneId / "settings" / "ssl", zone.tlsLevel.asJson)
+
+  val setSecurityLevel: CloudflareSettingFunction =
+    zone ⇒ zoneId ⇒ (BaseUrl / "zones" / zoneId / "settings" / "security_level", zone.securityLevel.asJson)
+
+  val allSettings: Set[CloudflareSettingFunction] = Set(setTlsLevel, setSecurityLevel)
```

Comment:
When we want to add more settings, we should create new `CloudflareSettingFunction` instances and add them to this list, and add the setting itself to `Zone`. This lets `ZoneSettingsClientImpl` coordinate making concurrent requests of the Cloudflare API, accumulating any errors that occur in a single response.